File: src/main/java/org/pyjjs/scheduler/core/api/impl/actors/resource/behaviours/ResourceInitBehaviour.java
package org.pyjjs.scheduler.core.api.impl.actors.resource.behaviours;

import org.pyjjs.scheduler.core.api.impl.actors.common.behaviours.Behaviour;
import org.pyjjs.scheduler.core.api.impl.actors.resource.ResourceActorState;
import org.pyjjs.scheduler.core.api.impl.actors.resource.supervisor.messages.ResourceInitMessage;

public class ResourceInitBehaviour extends Behaviour<ResourceActorState, ResourceInitMessage> {

    @Override
    protected void perform(ResourceInitMessage message) {
        ResourceActorState resourceActorState = getActorState();
        resourceActorState.setSource(message.getSource());
        resourceActorState.setInitialized(true);
        saveActorState(resourceActorState);
    }

    @Override
    protected Class<ResourceInitMessage> processMessage() {
        return ResourceInitMessage.class;
    }

}

File: src/main/kotlin/org/pyjjs/scheduler/core/api/impl/actors/resource/behaviours/ResourceInitBehaviour.kt
package org.pyjjs.scheduler.core.api.impl.actors.resource.behaviours

import org.pyjjs.scheduler.core.api.impl.actors.common.behaviours.Behaviour
import org.pyjjs.scheduler.core.api.impl.actors.common.messages.ResourceInitMessage
import org.pyjjs.scheduler.core.api.impl.actors.resource.ResourceActorState
import kotlin.reflect.KClass

class ResourceInitBehaviour : Behaviour<ResourceActorState, ResourceInitMessage>() {

    override fun perform(message: ResourceInitMessage) {
        val resourceActorState = actorState
        resourceActorState.source = message.source
        resourceActorState.timeSheet = message.source.timeSheet
        resourceActorState.isInitialized = true
        saveActorState(resourceActorState)
    }

    override fun processMessage(): KClass<ResourceInitMessage> {
        return ResourceInitMessage::class
    }

}
